builtin/providers/aws/resource_aws_ecs_service.go

```diff
@@ -510,6 +510,8 @@ func resourceAwsEcsServiceDelete(d *schema.ResourceData, meta interface{}) error
 func resourceAwsEcsLoadBalancerHash(v interface{}) int {
 	var buf bytes.Buffer
 	m := v.(map[string]interface{})
+
+	buf.WriteString(fmt.Sprintf("%s-", m["target_group_arn"].(string)))
```

Comment:
I can make that change. I followed the convention with elb_name which is not set if it's an ALB. In which case you will get empty and the hash should not change.

Anyway, I will verify and see what can be done to harden this a bit.